src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentRefactoringService.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    public interface IDocumentRefactoringService : IWorkspaceService
+    {
+        /// <summary>
+        /// Updates the contents of a document after some document info change, such as file name or path. May
+        /// prompt a user for confirmation on the changes.
+        /// </summary>
+        /// <returns>
+        /// Solution with the updated <paramref name="current"/> document
+        /// </returns> 
+        Task<Solution> UpdateAfterInfoChangeAsync(Document current, Document previous, CancellationToken cancellation = default);
```

Comment:
1. `CancellationToken cancellationToken`
2. the order of args is very confusing to me.  

I'm confused by what's going on here.  We have an interface, but then we supply the impl of it?  what's the interface needed for?